docs/reference-metrics-opentelemetry.md

```diff
@@ -8,6 +8,153 @@ Pulsar exposes the following OpenTelemetry metrics.
 
 ## Broker
 
+### Topic Messaging metrics
+
```

Comment:
What's the metric naming convention of OTel? Should we put `count`/`size` suffix?

Reply:
There's some recommendations: https://opentelemetry.io/docs/specs/semconv/general/metrics/#use-count-instead-of-pluralization-for-updowncounters. I'll add the suffixes where suggested, sure!